include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -29,7 +29,7 @@ namespace engine
 namespace routing_algorithms
 {
 
-template <class DataFacadeT, class Derived> class BasicRoutingInterface
+template <class DataFacadeT> class BasicRoutingInterface
```

Comment:
I think this was used in a CRTP pattern before, but I'm not completely sure about this. cc @TheMarex 

From what I can see it's unused currently and can go :+1: